.pre-commit-hooks.yaml

```diff
@@ -23,3 +23,9 @@
   types_or: ["text"]
   files: \.(po|pot)$
   require_serial: false
+- id: oca-check-unused-python-file
+  name: Check unused python files
+  entry: oca-check-unused-python-file
+  language: python
+  types: [python]
+  files: (controllers|models|report|wizard)\.py$|tests\/test_.*\.py$
```

Comment:
We have `hooks.py` not considered here

There are cases where the developer use plural or singular names or even use the root path of the module to create files directly

Only be aware to discard known cases:
- `__init__.py`
- `migrations/`

Reply:
True, I missed `hooks.py`. 

Initially I went with the exclude route, only excluding `__init__.py`, `migrations/` and `__manifest__.py`, but if you run the hook against Odoo, you will find there are tons of false positives, for example the drivers for IoT are python files and I am not sure they need to be imported, and they are also in non standard paths.

https://github.com/odoo/odoo/blob/16.0/addons/hw_drivers/iot_handlers/drivers/KeyboardUSBDriver.py

That is just an example, so I decided to instead just focus on well known folders that we know you must include all files from, which feels cleaner than excluding a few well known folders that we know you don't have to include files from.